krkn/scenario_plugins/kubevirt_vm_outage/kubevirt_vm_outage_scenario_plugin.py

```diff
@@ -0,0 +1,335 @@
+            if e.status == 404:
+                logging.warning(f"VMI {name} not found in namespace {namespace}")
+                return None
+            else:
+                logging.error(f"Error getting VMI {name}: {e}")
+                raise
+        except Exception as e:
+            logging.error(f"Unexpected error getting VMI {name}: {e}")
+            raise
+            
+    def execute_scenario(self, config: Dict[str, Any], scenario_telemetry: ScenarioTelemetry) -> int:
+        """
+        Execute a KubeVirt VM outage scenario based on the provided configuration.
+        
+        :param config: The scenario configuration
+        :param scenario_telemetry: The telemetry object for recording metrics
+        :return: 0 for success, 1 for failure
+        """
+        try:
+            params = config.get("parameters", {})
+            vm_name = params.get("vm_name")
+            namespace = params.get("namespace", "default")
+            duration = params.get("duration", 60)
+            disable_auto_restart = params.get("disable_auto_restart", False)
+            
+            if not vm_name:
+                logging.error("vm_name parameter is required")
+                return 1
+                
+            logging.info(f"Starting KubeVirt VM outage scenario for VM: {vm_name} in namespace: {namespace}")
+            
+            if not self.validate_environment(vm_name, namespace):
+                return 1
+                
+            vmi = self.get_vmi(vm_name, namespace)
+            if not vmi:
+                logging.error(f"VMI {vm_name} not found in namespace {namespace}")
+                return 1
+                
+            self.original_vmi = vmi
+            logging.info(f"Captured initial state of VMI: {vm_name}")
+            
+            result = self.inject(vm_name, namespace, disable_auto_restart)
+            if result != 0:
+                return 1
+                
+            logging.info(f"Waiting for {duration} seconds before attempting recovery")
+            time.sleep(duration)
+            
+            result = self.recover(vm_name, namespace)
+            if result != 0:
+                return 1
+                
+            logging.info(f"Successfully completed KubeVirt VM outage scenario for VM: {vm_name}")
+            return 0
+            
+        except Exception as e:
+            logging.error(f"Error executing KubeVirt VM outage scenario: {e}")
+            log_exception(e)
+            return 1
+
+    def validate_environment(self, vm_name: str, namespace: str) -> bool:
+        """
+        Validate that KubeVirt is installed and the specified VM exists.
+        
+        :param vm_name: Name of the VM to validate
+        :param namespace: Namespace of the VM
+        :return: True if environment is valid, False otherwise
+        """
+        try:
+            # Check if KubeVirt CRDs exist
+            crd_list = self.k8s_client.list_custom_resource_definition()
```

Comment:
doesn't look like this function exists in [krkn-lib](https://github.com/search?q=repo%3Akrkn-chaos%2Fkrkn-lib%20list_custom_resource_definition&type=code)? Do you have a PR to add it? You might want to also pass the kubevirt.io type to it and do the looping in the krkn-lib function itself

Reply:
We do have the [custom resource client](https://github.com/krkn-chaos/krkn-lib/blob/7ab3b0588ba8ab2fac4ba3f0c0abe8ae9743b94d/src/krkn_lib/k8s/krkn_kubernetes.py#L87-L89) already configured in krkn-lib, I think you'll want to use that client instead of k8s_client on all of the function calls here